connection_test.go

```diff
@@ -765,3 +766,35 @@ func TestConnectTimeout(t *testing.T) {
 		close(testComplete)
 	})
 }
+
+func TestTosPriority(t *testing.T) {
+	ctx, cancel := NewContext(time.Second)
+	defer cancel()
+
+	opts1 := testutils.NewOpts().SetServiceName("s1").SetTosPriority(tos.LOWDELAY).NoRelay()
```

Comment:
Some of these questions are hard for me to answer, while running the tests with and without my TestTosPriority sometimes other tests fail.  It become hard to determine signal from noise when reading the other failed tests.  To get something working i've copied from other tests.  Mostly naive on my part.   This feedback here helps.  I send a new diff shortly